cdap-client-tests/src/test/java/co/cask/cdap/client/ApplicationClientTestRun.java

```diff
@@ -169,14 +169,15 @@ public void testAppUpdate() throws Exception {
     Id.Artifact artifactIdV1 = Id.Artifact.from(Id.Namespace.DEFAULT, artifactName, "1.0.0");
     Id.Artifact artifactIdV2 = Id.Artifact.from(Id.Namespace.DEFAULT, artifactName, "2.0.0");
     Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, "ProgramsApp");
-    try {
-      artifactClient.add(Id.Namespace.DEFAULT, artifactName,
-        Files.newInputStreamSupplier(createAppJarFile(ConfigurableProgramsApp.class)),
-        "1.0.0");
-      artifactClient.add(Id.Namespace.DEFAULT, artifactName,
-        Files.newInputStreamSupplier(createAppJarFile(ConfigurableProgramsApp2.class)),
-        "2.0.0");
 
+    artifactClient.add(Id.Namespace.DEFAULT, artifactName,
```

Comment:
The reason it's like this is because the api doesn't require the version, but will try and derive it from the manifest if you don't supply one.  Could have a convenience method that takes an id though.